Client/Extensions/String+Extension.swift

```diff
@@ -39,4 +39,38 @@ extension String {
         attributedString.addAttributes(boldFontAttribute, range: NSRange(range, in: self))
         return attributedString
     }
+
+    /// Creates an attributed string with specified parts of the original string bolded.
+    ///
+    /// - Parameters:
+    ///   - boldPartsOfString: An array of strings to be bolded within the original string.
+    ///   - initialFont: The initial font to be used for the non-bold parts of the attributed string.
+    ///   - boldFont: The font to be used for the bolded parts of the attributed string.
+    /// - Returns: An `NSAttributedString` with the specified parts bolded.
+    ///
+    /// - Note: This method takes an array of strings (`boldPartsOfString`) and makes those parts bold within the original string.
+    /// It returns an attributed string with the specified parts bolded, using the `boldFont` for those parts and the `initialFont` for the non-bold parts.
+    ///
+    /// - Example:
+    ///   ```
+    ///   let originalString = "This is an example."
+    ///   let boldParts = ["example"]
+    ///   let initialFont = UIFont.systemFont(ofSize: 16)
+    ///   let boldFont = UIFont.boldSystemFont(ofSize: 16)
+    ///   let attributedString = originalString.attributedText(boldPartsOfString: boldParts,
+    ///                                                        initialFont: initialFont,
+    ///                                                        boldFont: boldFont)
+    ///   ```
+    func attributedText(boldPartsOfString: [NSString],
```

Comment:
By using attributedText(boldString:font:) we would get something like this:
![1](https://github.com/mozilla-mobile/firefox-ios/assets/51127880/c1b27b9d-8401-4fb3-b219-8ac0377ad554)
- The design doesn't look exactly as it should.

I didn't want to change the existing method because it might cause problems with the design in other parts of the app. That's why I created a new method. This new method allows us to set a starting font size for the entire text and a different font style and size for the part of the text that should be bold. This way, we can achieve the desired design and also have better control over the text.

![2](https://github.com/mozilla-mobile/firefox-ios/assets/51127880/86eaebc5-cb14-4adf-968a-6f6977211095)

I can also write unit tests for this method.